### ISSUE TITLE
Stacktrace is not shown in new console output

### ISSUE THREAD

**Issue by MatFl** [gradle/gradle#2079]
With the new console output in 3.5 the _--stacktrace_ option is ignored if _--console plain_ is not also specified.

`gradlew myTaskThatFails --stacktrace`

produces

```
FAILURE: Build failed with an exception.
[...]
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

But

`gradlew myTaskThatFails --stacktrace --console plain`

shows the complete stacktrace:

```
FAILURE: Build failed with an exception.
[...]
* Try:
Run with --info or --debug option to get more log output.

* Exception is:
[...]
```

**Comment by eriwen** [comment]
@MatFl Thanks for the report. Does the behavior improve if you try v4.0 M2? 

Can you also tell me more about your build environment? OS, Project type (Android/Webapp/Java), etc?

**Comment by w25r** [comment]
I was unable to reproduce this in Mac or Windows 10.  Let's wait for @MatFl to give environment information.

**Comment by MatFl** [comment]
I discovered this when investigating #2077. This happens on my build server (Windows 2012 R2 x64) during an android build.

I just tried to reproduce it with a simple task:
```
task throwE {
    throw new Exception("test")
}
```
However, here the stacktrace is properly shown. Maybe this is just a side effect of #2077. Because with the exception occurring there, I can reproduce this issue every time.

**Comment by eriwen** [comment]
@MatFl That information helps me narrow the cause of #2077 a bit. Seems like the exception may be thrown _because it's trying to print the stack trace_. Perhaps they're something a bit unique about that trace or the environment or the combination. Thank you. I think you're right that this is related, but let's keep and eye on this in case we find any other causes

**Comment by blindpirate** [comment]
Looks like this has been fixed by #2077 and #2291 . @MatFl I'm to close this, please feel free to reopen it if any problems still exist.
